pay-api/src/pay_api/services/statement.py

```diff
@@ -289,6 +289,62 @@ def get_statement_invoices(statement_id: int) -> List[InvoiceModel]:
             .filter(StatementInvoicesModel.statement_id == statement_id)
         )
 
+    @staticmethod
+    def calculate_invoice_summaries_by_payment_method(invoice_ids: List[int],
+                                                      payment_method: str,
+                                                      statement_to_date: str = None):
+        """Calculate invoice summaries for a specific payment method using database aggregation."""
+        if not invoice_ids:
+            return {
+                "paid_total": 0.0,
+                "due_total": 0.0,
+                "total_summary": 0.0
+            }
+
+        if payment_method != PaymentMethod.EFT.value:
+            # For non-EFT: refund applies if paid == 0 and refund > 0
+            refund_condition = case(
+                (and_(InvoiceModel.paid == 0, InvoiceModel.refund > 0), InvoiceModel.refund),
+                else_=0
+            )
+        else:
+            # For EFT: refund applies if paid == 0 and refund > 0 and refund_date <= statement.to_date
+            if statement_to_date:
+                refund_condition = case(
+                    (and_(
+                        InvoiceModel.paid == 0,
+                        InvoiceModel.refund > 0,
+                        InvoiceModel.refund_date.isnot(None),
+                        InvoiceModel.refund_date <= func.cast(statement_to_date, db.Date)
+                    ), InvoiceModel.refund),
```

Comment:
payment_date already accounted for?

Reply:
this logic is from template:
https://github.com/bcgov/bcros-common/pull/161/files
<img width="819" height="794" alt="image" src="https://github.com/user-attachments/assets/0dee1b9a-f229-4553-abfd-fdb87190845b" />
I think template missed, I added check for payment_date right now.